engine/crates/engine/src/registry/federation.rs

```diff
@@ -0,0 +1,145 @@
+//! Registry structs for using the engine as an [apollo federation subgraph][1]
+//!
+//! [1]: https://www.apollographql.com/docs/federation/subgraph-spec
+
+use serde_json::{Map, Value};
+
+/// Federation details for a particular entity
+///
+/// There should be one instance of this for each MetaType that represents
+/// a federation entity.
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, Default)]
+pub struct FederationEntity {
+    keys: Vec<FederationKey>,
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub enum FederationResolver {
+    DynamoUnique,
```

Comment:
Do I understand this correctly, but we can only resolve data from dynamo with the subgraph in the MVP?

Reply:
That's all we can do in this PR - but I'm not done with the MVP yet.  Dynamo was just the easiest thing to add so I could test everything else that's in here.